### PR TITLE
Have op_table.lazy_select_query() return a string instead of a list

### LINKED ISSUE #509: Error in strsplit() on "head-only" queries
When running a "head-only" query, the following error is returned

`Error in strsplit(name, ".", fixed = TRUE) : non-character argument`

Drilling down, the error occurs in dbplyr.R line 113: https://github.com/r-dbi/bigrquery/blob/73781ba90477025653ef2d10ee825bee7f407133/R/dplyr.R#L113
op_table() eventually arrives at op_table.lazy_select_query(), which returns a list instead of a character string. https://github.com/r-dbi/bigrquery/blob/73781ba90477025653ef2d10ee825bee7f407133/R/dplyr.R#L192 returns a list instead of the name.

This error does not occur when explicitly calling collect()

```r
if (bq_testable()) {
  ds <- bq_test_dataset(name = "bigrquery_test")

  bq_mtcars <- bq_table(ds, "mtcars")
  bq_table_exists(bq_mtcars)
  
  bq_table_upload(bq_mtcars, mtcars)
  
  library(dplyr)
  con <- bigrquery::dbConnect(bigrquery::bigquery(),
                              bigint = "character",
                              project = Sys.getenv("BIGQUERY_TEST_PROJECT"))
  
  test <- tbl(con, "bigrquery_test.mtcars")
  
  # strsplit error
  test

  # no strsplit error
  collect(test)
  
  }
```

### PR DESCRIPTION
**Pull request by clente**
Fix #509 by returning a string instead of a list.

### CODE REVIEW

File: R/dplyr.R [hunk 1]
```diff
@@ -192,7 +192,7 @@ op_table.lazy_base_remote_query <- function(x, con) x$x
 op_table.lazy_select_query <- function(x, con) {
   if (!query_is_head_only(x)) return(NULL)
 
-  x$x
+  x$x$x
```
Comment (mgirlich) [inline R/dplyr.R:195]: I think the cleaner version would be to use

```suggestion
op_table(x$x)
```

**Comment by pedrocava** [comment]
Hey, how's this PR going? Anything I can do to help it hit CRAN?

**Comment by clente** [comment]
@mgirlich any update on this? Is there anything I can help with?

**Comment by mgirlich** [comment]
I'm not a maintainer. There's nothing I can do here, sorry.

**Comment by muschellij2** [comment]
Had the same issue, thanks for the PR.   I think this fix is fine, but I'm not sure what downstream effects it'd have.  Looking deeper, I think the error is at
https://github.com/r-dbi/bigrquery/blob/3f187b20cd566e4c27e42ae1124b893f3fc53bb2/R/dplyr.R#L113.  The reason I think that is that `lq` is defined, but never used, so I think the code change is supposed to be:
```r
name <- op_table(lq, con)
```
I'll send a PR and ref this one to see what makes more sense.

Currently, those who need to use this PR can do so via:
```r
remotes::install_github("r-dbi/bigrquery#512")
```

The other option

**Comment by pedrocava** [comment]
`basedosdados` solved it in a similar way @muschellij2 : https://github.com/basedosdados/mais/pull/1563

**Review by jennybc — COMMENTED** [review]
LGTM

I'll see if Hadley can bless it.

**Comment by jennybc** [comment]
@clente A quick tip for future PRs to tidyverse or r-lib packages. It's preferred to give a human-readable description in the PR title and put the "fixes #xyz" part as the first line of the description. The basic idea is to make it easier for someone who doesn't happen to know what issue 509 is about.

https://tidyverse.github.io/code-review/author/submitting.html#title

**Comment by jennybc** [comment]
This should probably get a NEWS bullet since a bunch of people have clearly banged their shins on this.

https://style.tidyverse.org/news.html

**Review by hadley — APPROVED** [review]
Would be nice to have a test to make sure this or a similar mistake doesn't re-occur in the future.

**Comment by clente** [comment]
@jennybc sorry for the bad title. I'll make sure to read the full Review Principles before making another PR.

Thanks for the review, @hadley!

**Comment by jennybc** [comment]
Thanks for polishing this off!

One last tip: In future, you might want to make the PR from a feature branch (vs. from `main`).

**Pull request merged**